python/src/uagents/context.py

```diff
@@ -45,6 +48,29 @@
 EventCallback = Callable[["Context"], Awaitable[None]]
 
 
+def extract_agent_address(destination: str) -> str:
+    """
+    Extract the agent address from the provided destination.
+
+    Args:
+        destination (str): The destination address to check and extract.
+
+    Returns:
+        str: The extracted agent address if valid, or None if not valid.
+    """
+
+    prefixes = [TESTNET_PREFIX, MAINNET_PREFIX, ""]
+    expected_length = 65
+
+    for prefix in prefixes:
+        if destination.startswith(prefix) and len(destination) == expected_length + len(
+            prefix
+        ):
+            return destination[len(prefix) :]
+
+    return None
```

Comment:
I guess the best approach for this kind of function that can have multiple possible inputs is to write tests for each of the possible inputs. Here's what I think are the possibilities:
```python
agent1blahblahblah -> agent1blahblahblah
some-prefix://agent1blahblahblah -> agent1blahblahblah
some-prefix://some_agent_name/agent1blahblahblah -> agent1blahblahblah
```
This functions looks like it would work on the first two, but probably not on the third. I realize we haven't implemented the resolver logic for agent names in the identifier yet, so just looking a bit ahead here.

For these 3 cases, I think something as simple as:
```python
return destination.split("://")[-1].split("/")[-1]
```
could work, but needs to be tested.

Reply:
Ok I guess I didnt consider the other scenarios. Ill try this out